src/bmp.imageio/bmpinput.cpp

```diff
@@ -157,16 +157,17 @@ BmpInput::read_native_scanline(int subimage, int miplevel, int y, int z,
         for (unsigned int i = 0; i < m_spec.scanline_bytes();
              i += m_spec.nchannels)
             std::swap(fscanline[i], fscanline[i + 2]);
-        memcpy(data, &fscanline[0], m_spec.scanline_bytes());
+        memcpy(data, fscanline.get(), m_spec.scanline_bytes());
         return true;
     }
 
-    std::vector<unsigned char> mscanline(m_spec.scanline_bytes());
+    size_t scanline_bytes = m_spec.scanline_bytes();
+    std::unique_ptr<unsigned char[]> mscanline(new unsigned char[scanline_bytes]);
```

Comment:
Are you sure you need this copy? At the end, `mscanline` goes straight into `data`. Am I missing something subtle?

Reply:
No, I'm not sure, you're probably right. I was just chasing the low-hanging fruit on these second tier to make them 64 bit safe and, in a couple of spots, making the most obvious perf improvements when I didn't have to think about it hard.

Reply:
Yeah totally unrelated to this review - just looked like another easy potential speedup.

Reply:
When I was testing enormous 64k x 64k images, some formats took seconds to do a write/read cycle, others took minutes. In a couple of the latter instances that were most egregious, I spent a few minutes looking for obvious inefficiencies that could be fixed with just a couple lines and not much thought, just because it grated on me that this test overall took too long to run to be incorporated into the default CI suite. But it was a struggle, because I also knew that the bad ones were file formats that we never use, and that would never plausibly be used for images of that size, so it didn't really matter. Does BMP need to be fast? I think the answer is a clear no. But the perfectionist in me is upset by the inefficiency and needs to be pinned down long enough let the feeling pass and move on to things that actually matter.

Formats using RLE compression are the worst -- separate decisions and branching on every, inherently serial, can't easily be multithreaded or take advantage of SIMD.